File: level1JS/delDupleNum.js
function solution(arr) {
    const answer = [];
    arr.forEach((num, index) => {
        if(arr.length < index + 1) return;
        if(num !== arr[index + 1]) {
            answer.push(num);
        }
    })
    return answer;
}

console.log(solution([1,1,3,3,0,1,1]))

File: level1JAVA/src/DelDupleNum.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class DelDupleNum {
    public int[] solution(int[] arr) {
        List<Integer> answer = new ArrayList<>();

        int targetNum = -1;
        for(int num : arr) {
            if(targetNum == num) continue;
            answer.add(num);
            targetNum = num;
        }
        return answer.stream().mapToInt(i -> i).toArray();
    }

    public static void main(String[] args) {
        DelDupleNum delDupleNum = new DelDupleNum();
        int[] test = new int[]{1,1,3,3,0,1,1};
        int[] result = delDupleNum.solution(test);
        System.out.println("result = " + Arrays.toString(result));
    }
}
